tests/IBTK/marker_points_02.cpp

```diff
@@ -0,0 +1,196 @@
+        ++level_number;
+    }
+
+    // Set up marker points.
+    EigenAlignedVector<IBTK::Point> positions;
+    EigenAlignedVector<IBTK::Vector> velocities;
+    for (unsigned int i = 0; i < 10; ++i)
+    {
+        for (unsigned int j = 0; j < 10; ++j)
+        {
+            positions.emplace_back(double(i) / 10.0, double(j) / 10.0);
+            if (set_velocity)
+                velocities.emplace_back(std::numeric_limits<double>::max(), std::numeric_limits<double>::max());
+            else
+                velocities.emplace_back(double(i) * 10 + j, 0.0);
+        }
+    }
+    MarkerPoints marker_points("MarkerPoints", patch_hierarchy, positions, velocities);
+
+    int u_idx = IBTK::invalid_index;
+    if (set_velocity)
+    {
+        // Set up the velocity on the Cartesian grid:
+        Pointer<hier::Variable<NDIM> > u_var = new pdat::SideVariable<NDIM, double>("u_sc", 1);
+        const std::string kernel = test_db->getStringWithDefault("kernel", "PIECEWISE_LINEAR");
+        const int ghost_width = LEInteractor::getMinimumGhostWidth(kernel);
+
+        auto* var_db = hier::VariableDatabase<NDIM>::getDatabase();
+        tbox::Pointer<hier::VariableContext> ctx = var_db->getContext("context");
+        u_idx = var_db->registerVariableAndContext(u_var, ctx, IntVector<NDIM>(ghost_width));
+
+        for (int ln = 0; ln <= patch_hierarchy->getFinestLevelNumber(); ++ln)
+        {
+            tbox::Pointer<hier::PatchLevel<NDIM> > level = patch_hierarchy->getPatchLevel(ln);
+            level->allocatePatchData(u_idx, 0.0);
+        }
+        IBTK::muParserCartGridFunction u_fcn("u", test_db->getDatabase("u"), patch_hierarchy->getGridGeometry());
+        u_fcn.setDataOnPatchHierarchy(u_idx, u_var, patch_hierarchy, 0.0);
+
+        std::vector<std::unique_ptr<muParserRobinBcCoefs> > u_bc_coefs;
+        std::vector<RobinBcCoefStrategy<NDIM>*> u_bc_coef_ptrs;
+
+        if (test_db->keyExists("UBcCoefs_0"))
+        {
+            for (int d = 0; d < NDIM; ++d)
+            {
+                u_bc_coefs.emplace_back(new muParserRobinBcCoefs("u_bc_coefs_" + std::to_string(d),
+                                                                 test_db->getDatabase("UBcCoefs_" + std::to_string(d)),
+                                                                 grid_geometry));
+                u_bc_coef_ptrs.push_back(u_bc_coefs.back().get());
+            }
+        }
+
+        // sync ghost data:
+        using ITC = IBTK::HierarchyGhostCellInterpolation::InterpolationTransactionComponent;
+        std::vector<ITC> ghost_cell_components(1);
+        ghost_cell_components[0] = ITC(
+            u_idx, "CONSERVATIVE_LINEAR_REFINE", true, "CONSERVATIVE_COARSEN", "LINEAR", true, u_bc_coef_ptrs, nullptr);
+        IBTK::HierarchyGhostCellInterpolation ghost_fill_op;
+        ghost_fill_op.initializeOperatorState(ghost_cell_components, patch_hierarchy);
+        ghost_fill_op.fillData(/*time*/ 0.0);
```

Comment:
There's something slightly wrong with this ghostfilling approach: some corners are not filled correctly. I rigged up the tests to avoid that problem but it would still be nice to fix this.